mindsdb/integrations/handlers/influxdb_handler/influxdb_handler.py

```diff
@@ -0,0 +1,232 @@
+        Args:
+            name (str): name of particular handler instance
+            connection_data (dict): parameters for connecting to the database
+            **kwargs: arbitrary keyword arguments.
+        """
+        super().__init__(name)
+        self.parser = parse_sql
+        self.dialect = 'influxdb'
+        self.connection_data = connection_data
+        self.kwargs = kwargs
+
+        self.connection = None
+        self.is_connected = False
+
+    def __del__(self):
+        if self.is_connected is True:
+            self.disconnect()
+
+    def connect(self) -> StatusResponse:
+        """
+        Set up the connection required by the handler.
+        Returns:
+            HandlerStatusResponse
+        """
+
+        if self.is_connected is True:
+            return self.connection
+        
+        url = f"{self.connection_data['influxdb_url']}/query"
+        params = {
+            ('db',f"{self.connection_data['influxdb_db_name']}"),
+            ('q',f"{self.connection_data['influxdb_query']}" )
+        }
+        headers = {
+            "Authorization": f"Token {self.connection_data['influxdb_token']}",
+            "Accept": "application/csv",
+        }
+        response = requests.request("GET",url,params=params,headers=headers)
+        
+        if response.status_code == 200:
+            self.connection = response
+
+        self.is_connected = True
+
+        return self.connection
+
+    def disconnect(self):
+        """
+        Close any existing connections.
+        """
+
+        if self.is_connected is False:
+            return
+
+        self.connection.close()
+        self.is_connected = False
+        return self.is_connected
+
+    def check_connection(self) -> StatusResponse:
+        """
+        Check connection to the handler.
+        Returns:
+            HandlerStatusResponse
+        """
+
+        response = StatusResponse(False)
+        need_to_close = self.is_connected is False
+
+        try:
+            self.connect()
+            response.success = True
+        except Exception as e:
+            log.error(f'Error connecting to influxdb portal {self.connection_data["influxdb_url"]}, {e}!')
+            response.error_message = str(e)
+        finally:
+            if response.success is True and need_to_close:
+                self.disconnect()
+            if response.success is False and self.is_connected is True:
+                self.is_connected = False
+
+        return response
+
+    def native_query(self, query: str) -> StatusResponse:
+        """
+        Receive raw query and act upon it somehow.
+        Args:
+            query (str): query in native format
+        Returns:
+            HandlerResponse
+        """
+
+        need_to_close = self.is_connected is False
+
+        connection = self.connect()
+        df = pd.read_csv(io.StringIO(connection.text))
+        locals()[self.connection_data['influxdb_table_name']] = df
```

Comment:
Do we need to use locals?

Reply:
I am using locals to change the  string to variable